content/vgg-classifier-tutorial.md

```diff
@@ -0,0 +1,320 @@
+Here we are using the VGG16 model, if you want more information, please read: http://arxiv.org/abs/1409.1556.
+
+Next, you have to create a ``Keras Sequential`` model (just a stack of layers) with the respect of the original model topology of ``vgg16_weights.h5``.
+
+```python
+def VGG16(weights_path=None):
+	model = Sequential()
+	model.add(ZeroPadding2D((1,1),input_shape=(3,224,224)))
+	model.add(Convolution2D(64, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(64, 3, 3, activation='relu'))
+	model.add(MaxPooling2D((2,2), strides=(2,2)))
+
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(128, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(128, 3, 3, activation='relu'))
+	model.add(MaxPooling2D((2,2), strides=(2,2)))
+
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(256, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(256, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(256, 3, 3, activation='relu'))
+	model.add(MaxPooling2D((2,2), strides=(2,2)))
+
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(MaxPooling2D((2,2), strides=(2,2)))
+
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(ZeroPadding2D((1,1)))
+	model.add(Convolution2D(512, 3, 3, activation='relu'))
+	model.add(MaxPooling2D((2,2), strides=(2,2)))
+
+	model.add(Flatten())
+	model.add(Dense(4096, activation='relu'))
+	model.add(Dropout(0.5))
+	model.add(Dense(4096, activation='relu'))
+	model.add(Dropout(0.5))
+	model.add(Dense(1000, activation='softmax'))
+...
+```
+Then you can load the model:
+```python
+...
+	if weights_path:
+		print("Loading weights:",weights_path)
+		model.load_weights(weights_path)
+
+	return model
+
+mymodel = VGG16(weights_path='vgg16_weights.h5')
+
+```
+
+Modify VGG for your own dataset
+-------
+As you can see on the original VGG16 model train on ImageNet, the output number of classes is ``1000``. But you have to change it for your own dataset. To change it, you just need to change the last ``Dense`` (fully connected) layer of the model to fit with our dataset number of classes (``101`` in this tutorial).
+
+### Change the last layer
+For changing the number of classes of your model, you need to redefine the layer:
+
+```python
+mymodel.layers[-1] = Dense(101, activation='softmax'))
+```
+and reattached it to the before last layer:
+
+```python
+mymodel.layers[-1].set_previous(pretrained_model.layers[-2])
```

Comment:
This will not work with the new Keras. I am 99% sure that the following would work, although it needs to be tested"

```python
del model.layers[-1]
model.add(Dense(101, activation='softmax'))
```

I would suggest deleting all FC layers are replacing them with new, smaller FC layers (adapted to a small dataset).